Finance/Stonks/CurrencyData.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace Finance
+{
+    public class CurrencyData
+    {
+        public int Cur_ID { get; set; }
```

Comment:
вы преобразуете string в объект этого класса с помощью `System.Text.Json.JsonSerializer`

С# свойство можно назвать согласно код-конвенции языка, а правильная десериализация будет обеспена атрибутом:
```suggestion
        [JsonPropertyName("Cur_ID ")]
        public int CurId { get; set; }
```